rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-autoscaling.adoc

```diff
@@ -0,0 +1,85 @@
+:_mod-docs-content-type: ASSEMBLY
+[id="cloud-experts-getting-started-autoscaling"]
+= Tutorial: Autoscaling
+include::_attributes/attributes-openshift-dedicated.adoc[]
+:context: cloud-experts-getting-started-autoscaling
+
+toc::[]
+
+//rosaworkshop.io content metadata
+//Brought into ROSA product docs 2024-01-04
+
+The xref:../../rosa_cluster_admin/rosa_nodes/rosa-nodes-about-autoscaling-nodes.adoc#rosa-nodes-about-autoscaling-nodes[cluster autoscaler] adds or removes worker nodes from a cluster based on pod resources.
+
+The cluster autoscaler increases the size of the cluster when:
+
+* Pods fail to schedule on the current nodes due to insufficient resources.
+* Another node is necessary to meet deployment needs.
+
+The cluster autoscaler does not increase the cluster resources beyond the limits that you specify. 
+
+The cluster autoscaler decreases the size of the cluster when:
+
+* Some nodes are consistently not needed for a significant period. For example, when a node has low resource use and all of its important pods can fit on other nodes.
+
+== Enabling autoscaling for an existing machine pool using the CLI
+
+[NOTE]
+====
+Cluster autoscaling can be enabled at cluster creation and when creating a new machine pool by using the `--enable-autoscaling` option.
+====
+
+. Autoscaling is set based on machine pool availability. To find out which machine pools are available for autoscaling, run the following command:
++
+[source,terminal]
+----
+$ rosa list machinepools -c <cluster-name>
+----
++
+.Example output
++
+[source,terminal]
+----
+ID         AUTOSCALING  REPLICAS  INSTANCE TYPE  LABELS     TAINTS    AVAILABILITY ZONES
+Default    No           2         m5.xlarge                           us-east-1a
+----
+
+. Run the following command to add autoscaling to an available machine pool:
++
+[source,terminal]
+----
+$ rosa edit machinepool -c <cluster-name> --enable-autoscaling <machinepool-name> --min-replicas=<num> --max-replicas=<num>
+----
++
+.Example input
++
+[source,terminal]
+----
+$ rosa edit machinepool -c my-rosa-cluster --enable-autoscaling Default --min-replicas=2 --max-replicas=4
+----
++
+The above command creates an autoscaler for the worker nodes that scales between 2 and 4 nodes depending on the resources.
+
+== Enabling autoscaling for an existing machine pool using the UI
+
+[NOTE]
+====
+Cluster autoscaling can be enabled at cluster creation by checking the *Enable autoscaling* checkbox when creating machine pools.
+====
+
+. Go to the *Machine pools* tab and click the three dots in the right..
+. Click *Scale*, then *Enable autoscaling*.
```

Comment:
It should be "edit" and no scale option available in UI.
![Screenshot from 2024-09-24 11-48-58](https://github.com/user-attachments/assets/833134b7-fcec-40e2-a6a6-89242f0ed45d)

Reply:
After enable autoscaling option, user could update the minimum node count and maximum node count or leave the default values in both fields.
Finally user has to save the definition to take effect of changes in machine pool.